source/solvers/cahn_hilliard_scratch_data.cc

```diff
@@ -65,6 +65,66 @@ CahnHilliardScratchData<dim>::allocate()
     std::vector<Tensor<1, dim>>(this->n_q_points));
   this->velocity_gradient_values =
     std::vector<Tensor<2, dim>>(this->n_q_points);
+
+  // Allocate physical properties
+  this->surface_tension                 = std::vector<double>(n_q_points);
+  this->mobility_cahn_hilliard          = std::vector<double>(n_q_points);
+  this->mobility_cahn_hilliard_gradient = std::vector<double>(n_q_points);
+
+  // Physical properties
+  fields.insert(
+    std::pair<field, std::vector<double>>(field::phase_order_cahn_hilliard,
+                                          n_q_points));
+  fields.insert(std::pair<field, std::vector<double>>(
+    field::phase_order_cahn_hilliard_filtered, n_q_points));
+}
+
+template <int dim>
+void
+CahnHilliardScratchData<dim>::calculate_physical_properties()
+{
+  switch (properties_manager.get_number_of_fluids())
+    {
+      case 1:
+        {
+          throw std::runtime_error("Unsupported number of fluids (<2)");
+        }
+      case 2:
+        {
+          set_field_vector(field::phase_order_cahn_hilliard,
+                           this->phase_order_values,
+                           this->fields);
+
+          // Gather properties from material interactions if necessary
+          if (properties_manager.get_number_of_material_interactions() > 0)
+            {
+              const auto material_interaction_id =
+                properties_manager.get_material_interaction_id(
+                  material_interactions_type::fluid_fluid, 0, 1);
+              // Gather surface tension
+              const auto surface_tension_model =
+                properties_manager.get_surface_tension(material_interaction_id);
+              surface_tension_model->vector_value(fields, surface_tension);
+            }
+
+          // Gather surface tension
+          const auto material_interaction_id =
+            properties_manager.get_material_interaction_id(
+              material_interactions_type::fluid_fluid, 0, 1);
+
```

Comment:
For CH simulations, the number of material interactions is always 1 or over.

```suggestion
          const auto material_interaction_id =
            properties_manager.get_material_interaction_id(
              material_interactions_type::fluid_fluid, 0, 1);

          const auto surface_tension_model =
            properties_manager.get_surface_tension(material_interaction_id);
          surface_tension_model->vector_value(fields, surface_tension);
```

Reply:
agreed